### ISSUE TITLE
New functionality for rules

### ISSUE THREAD

**Issue by YarLeo** [stuartpittaway/diyBMSv4Code#16]
Hi @stuartpittaway ,
Here are my suggestions about extending the rules functionality (move here from the community on your request).

Long story short:
1) Add "Individual cell module over temperature (internal) °C"
2) Add second "Individual cell over temperature (external) °C" for High and High High signals
3) "Pack under voltage (mV)" - divide into 2 set points: under voltage + recover set point
4) "Pack over voltage (mV)" - divide into 2 set points: over voltage + recover set point

Detailed description with examples:
1) Add an internal over temperature rule to start cooling fan for the bms. Example: cells modules are installed separately from cells, all together, lets say in a box. With such installation it is easy to organize a cooling for them to make top balancing more efficient for large packs. Also in such case the interface/communication wires are as short as possible to minimize interference. A rule to start cooling would be nice to have in this case. 

2) Individual cell over temperature (external) °C - it would be great to have this rule 2 times to configure HIGH and HIGH HIGH limits. HIGH limit would be used to switch on cooling fans in the battery compartment and HIGH HIGH - as a main trip signal to disconnect the battery from all devices.

3) Pack under voltage. The most reasonable use for this rule is disconnecting and connecting back to the battery inverters. I saw somebody already suggested to put a time delay for rules, but I would like to extend this subject. Normally for such application 2 set points are used: switch off set point and recover set point. Time delay would also be great (to cover start loads) but not enough, from my point of view. Example: 14s battery + grid tie inverter with limiter - the inverter itself will normally have a battery discharge protection, however it make no sense to “dry run” an inverter if the battery is discharged, because it will only consume power from grid (yes, not much, but still); so, it would be nice to switch off the inverter lets say at 44V and connect it back only after battery is recharged to lets say 49V. Using this rule “as it is” with only one set point make for me little sense, because it would be jumping back and forth all the time.

4) Pack over voltage (mV) - similar thing as case 3 but for burning the excess energy (to heat the water for example). We can switch on the relay to turn on the water heater at Lets say 57,5V and switch it off at 56V. A time delay would also be nice here.

It would be nice to have these features in new releases.
Thanks

**Comment by stuartpittaway** [comment]
These changes have been made as part of the new relay rules.  Please review - in master branch.
